.editorconfig

```diff
@@ -0,0 +1,12 @@
+# EditorConfig is awesome: http://EditorConfig.org
+
+# top-most EditorConfig file
+root=true
+
+[*]
+charset=utf-8
+# Unix-style newlines with a newline ending every file
+end_of_line=lf
+insert_final_newline=true
+indent_style=tab
+tab_width=4
```

Comment:
This is not related to gradle, please revert this change.

Reply:
OK, will make a separate PR

Reply:
I don't think we need to clutter the repo with this.
Every sane editor detects line-endings and tab/whitespace automatically anyway.

Reply:
I understand you but I have a sane Android Studio editor, and even though it automatically detects coding conventions already present in the sources and keeps my changes consistent with them, it pops a yellow warning bar on top of the text editor telling me that it replaced spaces with tabs to follow the document's formatting style but that does not match my editor preferences.
.editorconfig is the clean way to avoid any such inconsistencies and tell any sane editor what are the project's preferences that should override any globally configured user's preferences.
I truly believe you should not resist this clean and standard way to achieve this consistency and you have to admit that this "clutter" is truly minimal and so not worth avoiding.
By the way, presence of .editorconfig would even make it unnecessary to mention the corresponding formatting conventions in the Readme (which it currently does).